### ISSUE TITLE
`any` refs with `:std:term` targets crash when an ambiguity is encountered

### ISSUE THREAD

**Issue by kohr-h** [sphinx-doc/sphinx#4662]
Since 1.7, `any` targets raise a `KeyError` in [this line](https://github.com/sphinx-doc/sphinx/blob/f8bb40ab308b5daacfe45979a4e42fa82f8ef8ed/sphinx/transforms/post_transforms/__init__.py#L138) since apparently, the `Element` object in question has no `'reftitle'` attribute.

[Here](https://travis-ci.org/odlgroup/odl/jobs/344544774)'s a link to a failing Travis build, although the error message isn't really that exhaustive. If I build locally, I get this traceback in the error log:
```
Traceback (most recent call last):
  File "[...]/lib/python3.6/site-packages/sphinx/cmdline.py", line 304, in main
    app.build(args.force_all, filenames)
  File "[...]/lib/python3.6/site-packages/sphinx/application.py", line 330, in build
    self.builder.build_update()
  File "[...]/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 333, in build_update
    'out of date' % len(to_build))
  File "[...]/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 394, in build
    self.write(docnames, list(updated_docnames), method)
  File "[...]/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 431, in write
    self._write_serial(sorted(docnames))
  File "[...]/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 438, in _write_serial
    doctree = self.env.get_and_resolve_doctree(docname, self)
  File "[...]/lib/python3.6/site-packages/sphinx/environment/__init__.py", line 813, in get_and_resolve_doctree
    self.apply_post_transforms(doctree, docname)
  File "[...]/lib/python3.6/site-packages/sphinx/environment/__init__.py", line 860, in apply_post_transforms
    transformer.apply_transforms()
  File "[...]/lib/python3.6/site-packages/sphinx/transforms/__init__.py", line 96, in apply_transforms
    Transformer.apply_transforms(self)
  File "[...]/lib/python3.6/site-packages/docutils/transforms/__init__.py", line 171, in apply_transforms
    transform.apply(**kwargs)
  File "[...]/lib/python3.6/site-packages/sphinx/transforms/post_transforms/__init__.py", line 93, in apply
    newnode = self.resolve_anyref(refdoc, node, contnode)
  File "[...]/lib/python3.6/site-packages/sphinx/transforms/post_transforms/__init__.py", line 139, in resolve_anyref
    for name, role in results)
  File "[...]/lib/python3.6/site-packages/sphinx/transforms/post_transforms/__init__.py", line 139, in <genexpr>
    for name, role in results)
  File "[...]/lib/python3.6/site-packages/docutils/nodes.py", line 572, in __getitem__
    raise e
  File "[...]/lib/python3.6/site-packages/docutils/nodes.py", line 568, in __getitem__
    return self.attributes[key]
KeyError: 'reftitle'
```

When I replace the raising lookup in `post_transforms` with `role.get('reftitle', '<none>')` I see that AFAICT only the `:std:term` targets are affected, and the build runs normally otherwise. Warnings are of the type
```
WARNING: more than one target found for 'any' cross-reference 'Operator': could be :std:term:`<none>` or :py:class:`odl.operator.operator.Operator`
```

### Environment info
- OS: Linux
- Python version: 3.6
- Sphinx version: 1.7.0

**Comment by tk0miya** [comment]
Is there any procedure to reproduce?

**Comment by kohr-h** [comment]
Well, there may be, but I can't think of an easy way to generate a minimal example. If you know any, or if you can point me to a unit test that I could look into in order to test handling of glossary terms, I'd be grateful to learn.

**Comment by tk0miya** [comment]
Fixed by #4666 
Thank you for reporting.

**Comment by kohr-h** [comment]
Oh great, thanks a lot @tk0miya!
